src/mdns/mdns_mdnssd.cpp

```diff
@@ -415,9 +462,133 @@ otbrError PublisherMDnsSd::PublishService(uint16_t aPort, const char *aName, con
     return ret;
 }
 
-Publisher *Publisher::Create(int aFamily, const char *aHost, const char *aDomain, StateHandler aHandler, void *aContext)
+otbrError PublisherMDnsSd::UnpublishService(const char *aName, const char *aType)
+{
+    DiscardService(aName, aType);
+    return OTBR_ERROR_NONE;
+}
+
+otbrError PublisherMDnsSd::PublishHost(const char *aName, const uint8_t *aAddress, uint8_t aAddressLength)
+{
+    otbrError    ret   = OTBR_ERROR_NONE;
+    int          error = 0;
+    char         fullName[kMaxSizeOfDomain];
+    DNSRecordRef hostRecord;
+    Host         newHost;
+
+    SuccessOrExit(ret = MakeFullName(fullName, sizeof(fullName), aName));
+
+    if (mHostsConnection == nullptr)
+    {
+        SuccessOrExit(error = DNSServiceCreateConnection(&mHostsConnection));
+    }
+
+    for (const auto &host : mHosts)
+    {
+        if (strcmp(aName, host.mName) == 0)
+        {
+            otbrLog(OTBR_LOG_INFO, "mDNS update host %s", aName);
+            SuccessOrExit(error = DNSServiceUpdateRecord(mHostsConnection, host.mRecord, /* flags */ 0, aAddressLength,
+                                                         aAddress, /* ttl */ 0));
+            ExitNow();
+        }
+    }
+
+    SuccessOrExit(error = DNSServiceRegisterRecord(mHostsConnection, &hostRecord, kDNSServiceFlagsUnique,
+                                                   kDNSServiceInterfaceIndexAny, fullName, kDNSServiceType_AAAA,
```

Comment:
Will using `kDNSServiceInterfaceIndexAny` cause dns-sd to publish records with multicasts at wpan0 as well?

Reply:
If there are issues with multicast on `wpan0`, then there should already been issues with calling `DNSServiceRegister()`.

Reply:
Agreed to resolve them in the OpenThread posix firewall.